moto/core/exceptions.py

```diff
@@ -62,9 +63,8 @@ def __init__(
         self.error_type = error_type
         self.message = message
 
-        if template in self.templates.keys():
-            env = Environment(loader=DictLoader(self.templates))
-            self.description: str = env.get_template(template).render(
+        if template in self.env.list_templates():
```

Comment:
Since templates isn't updated only `self.env` is (changed when adding `extended_environment` to make the code less repetitive